### ISSUE TITLE
setToken redirect the browser even if redirect param is set to false

### ISSUE THREAD

**Issue by mbellezi** [sahat/satellizer#326]
The method setToken is redirecting even if the redirect param is set to false as below:

                $auth.setToken(data.token, false);

**Comment by matt-zangen** [comment]
Have you blanked out $authProvider.loginRedirect in the config phase of your app?

[The default value for this config variable is '/'](https://github.com/sahat/satellizer#configuration), which will trigger [this redirect check regardless of a false redirect in setToken](https://github.com/sahat/satellizer/blob/bc4a59e595e462353d73b52c39d965141bccfd77/satellizer.js#L309):

    if (config.loginRedirect && !redirect) {
        $location.path(config.loginRedirect);
    ...

Check the [example app's](https://github.com/sahat/satellizer/blob/bc4a59e595e462353d73b52c39d965141bccfd77/examples/client/app.js#L2) for an example config block, where you would simply add the following:

    $authProvider.loginRedirect = '';

**Comment by mbellezi** [comment]
Ok. If I set it to blank it works. 
But, what is the point to have this redirect param?
For example, in my app, when I update the user profile information I receive a new token with the updated user information and store it using setToken. In this case I do not need the redirection because the user is already logged in.
If this code fragment is modified in this way:

    if (config.loginRedirect && redirect) {
        $location.path(config.loginRedirect);
    ...

I think it will obey this redirec param.
What do you think?

**Comment by matt-zangen** [comment]
[Follow this link to see the full conditional](https://github.com/sahat/satellizer/blob/bc4a59e595e462353d73b52c39d965141bccfd77/satellizer.js#L309) is as follows:

    if (config.loginRedirect && !redirect) {
        $location.path(config.loginRedirect);
    }  else if (redirect && angular.isString(redirect)) {
        $location.path(encodeURI(redirect));
    }

This structure prioritizes redirect when it's supplied, but when it isn't, it will fall back on config.loginRedirect for default. The change you're suggesting would only apply the default redirect if an explicit redirect is also supplied, breaking this example:

    // refresh token and redirect home (default)
    auth.setToken(token);

If I'm understanding your goal, you want to rely on config.loginRedirect normally, but want to be able to explicitly suppress redirection in only certain scenarios like yours, when you want a new token with no redirect. Is that right?

**Comment by mbellezi** [comment]
I see.
Yes, that was my goal.
I've misunderstood the current meaning of this redirect param. Sorry for that.
Another try:

    if (config.loginRedirect && !redirect && (redirect !== false)) {
        $location.path(config.loginRedirect);
    }  else if (redirect && angular.isString(redirect)) {
        $location.path(encodeURI(redirect));
    }

**Comment by matt-zangen** [comment]
That solution makes sense and I think the problem is valid enough to submit a PR. It's not an ideal situation, though. This is a workaround, and the behavior would have to be specially documented.

I think this is showcasing the complexity of having satellizer involved in routing. Perhaps I should mention this in a separate Issue/PR, but I had a problem with redirects using $location instead of $state, since I'm using ui-router, and I had to share router config with satellizer's config to satisfy it. When routing gets baked-in like this, satellizer becomes opinionated on behavior outside of authentication, when redirects could easily be put into a promise resolution.

I think your suggested PR is valid for this issue, but it's my opinion that redirection should be removed entirely from satellizer. That's what promises are for.

**Comment by mbellezi** [comment]
I agree. Routing operations should be handled by the application as there are a number of router modules and this number is increasing. 
With the promise returned by the $http module it is very trivial to do the app specific routing.
Thank you.

**Comment by philippeluickx** [comment]
Relevant issue, especially considering the docs:


$auth.setToken(token, [redirect])

Saves a JWT or an access token to Local Storage. It uses shared.setToken internally.

token - An object that takes a JWT (response.data[config.tokenName]) or an access token (response.access_token).
redirect - An optional boolean value that controls whether or not to redirect to loginRedirect route after saving a token. Defaults to false.

IMO the docs make more sense then current behaviour. Which redirect should be used is maybe more something to decided in the actual login flow?

**Comment by JetFault** [comment]
Yeah, I agree with this. If you're looking for a temp workaround, set redirect to `true`. This way its not a string and it exists; skipping both redirect conditions.

Also, maybe make `authenticate` and `login` be the same. `login` takes a redirect param while authenticate does not and sends in `false`
